sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/GreaterThan.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.expressions.filter;
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.connector.expressions.Expression;
+import org.apache.spark.sql.connector.expressions.FieldReference;
+import org.apache.spark.sql.connector.expressions.NamedReference;
+
+/**
+ * A filter that evaluates to `true` iff the field evaluates to a value
+ * greater than `value`.
+ *
+ * @since 3.3.0
+ */
+@Evolving
+public final class GreaterThan extends Filter {
+  private final FieldReference column;
+  private final Expression value;
```

Comment:
This looks a bit too ambitious. How about `org.apache.spark.sql.connector.expressions.Literal`?

Reply:
We can have nested Filter such as `new EqualTo(ref("a"), new EqualTo(ref("b"), LiteralValue(1, IntegerType))` if we use `Expression`.

Reply:
You are right, shouldn't have a nested filter this way. Changed the type to `Literal`.